sandbox/samples/cpp/alembicassembly/alembicassembly.cpp

```diff
@@ -0,0 +1,1525 @@
+                floor_index,
+                ceil_index);
+
+            // now get floor and ceil samples
+            AbcGeom::CameraSample floor_sample;
+            AbcGeom::CameraSample ceil_sample;
+
+            cam_schema.get(floor_sample, floor_index);
+            cam_schema.get(ceil_sample, ceil_index);
+
+            // and interpolate sample camera values
+            const float mid_focal_length = asf::lerp<float>(
+                floor_sample.getFocalLength(),
+                ceil_sample.getFocalLength(),
+                alpha);
+
+            const float mid_f_stop = asf::lerp<float>(
+                floor_sample.getFStop(),
+                ceil_sample.getFStop(),
+                alpha);
+
+            const float mid_focal_distance = asf::lerp<float>(
+                floor_sample.getFocusDistance(),
+                ceil_sample.getFocusDistance(),
+                alpha);
+
+            const float mid_film_width = asf::lerp<float>(
+                floor_sample.getHorizontalAperture(),
+                ceil_sample.getHorizontalAperture(),
+                alpha);
+
+            const float mid_film_height = asf::lerp<float>(
+                floor_sample.getVerticalAperture(),
+                ceil_sample.getVerticalAperture(),
+                alpha);
+
+            // maya related
+            /*const float mid_aspect_ratio = asf::lerp<float>(
+                floor_sample.getLensSqueezeRatio(),
+                ceil_sample.getLensSqueezeRatio(),
+                alpha);*/
+
+            const float mid_near_z = asf::lerp<float>(
+                floor_sample.getNearClippingPlane(),
+                ceil_sample.getNearClippingPlane(),
+                alpha);
+
+            asr::ParamArray& cam_params = m_camera->get_parameters();
+
+            cam_params.insert("focal_length", mid_focal_length/1000.0f);  // to mm
+            cam_params.insert("f_stop", mid_f_stop);
+            cam_params.insert("focal_distance", mid_focal_distance);
+            cam_params.insert("film_width", mid_film_width / 100.0f);
+            cam_params.insert("film_height", mid_film_height / 100.0f);
+            cam_params.insert("near_z", -mid_near_z / 100.0f);
+
+            cam_params.insert("shutter_open_time", shutter_open);
+            cam_params.insert("shutter_close_time", shutter_close);
+
+            ///////////////////////////////////////////////////////////////////
+            // xform
+            ///////////////////////////////////////////////////////////////////
+            // generate xform sequence stack from our camera
+            const std::vector<asr::TransformSequence> xform_seq_stack = xform_seq_for_obj(
+                cam_obj,
+                shutter_open,
+                shutter_close,
+                time_offset);
+
+            if (xform_seq_stack.size())
+            {
+                const asr::TransformSequence xform_seq = flatten_xform_seq(xform_seq_stack);
```

Comment:
Get rid of temporary variable?

Reply:
Done.

I like to have one line doing one thing and let the compiler optimize. But in this case it's pretty obvious.